File: cdm-sources/cdm/product/template/functions/FpmlIrd8.java
package cdm.product.template.functions;

import cdm.base.staticdata.party.Account;
import cdm.event.common.Trade;
import com.google.inject.ImplementedBy;
import com.rosetta.model.lib.functions.RosettaFunction;
import java.util.Collections;
import java.util.List;


@ImplementedBy(FpmlIrd8.FpmlIrd8Default.class)
public abstract class FpmlIrd8 implements RosettaFunction {

	/**
	* @param trade 
	* @param accounts 
	* @return success Validation result
	*/
	public Boolean evaluate(Trade trade, List<? extends Account> accounts) {
		Boolean success = doEvaluate(trade, accounts);
		
		return success;
	}

	protected abstract Boolean doEvaluate(Trade trade, List<? extends Account> accounts);

	public static class FpmlIrd8Default extends FpmlIrd8 {
		@Override
		protected Boolean doEvaluate(Trade trade, List<? extends Account> accounts) {
			if (accounts == null) {
				accounts = Collections.emptyList();
			}
			Boolean success = null;
			return assignOutput(success, trade, accounts);
		}
		
		protected Boolean assignOutput(Boolean success, Trade trade, List<? extends Account> accounts) {
			return success;
		}
	}
}

File: cdm-sources/src/cdm/product/template/functions/FpmlIrd8.py
# pylint: disable=line-too-long, invalid-name, missing-function-docstring, missing-module-docstring, superfluous-parens
# pylint: disable=wrong-import-position, unused-import, unused-wildcard-import, wildcard-import, wrong-import-order, missing-class-docstring
from __future__ import annotations
import sys
import datetime
import inspect
from decimal import Decimal
from rosetta.runtime.utils import *
from rosetta.runtime.func_proxy import replaceable, create_module_attr_guardian
from cdm.base.staticdata.party.Account import Account
from cdm.event.common.Trade import Trade

__all__ = ['FpmlIrd8']


@replaceable
def FpmlIrd8(trade: Trade, accounts: list[Account] | None) -> bool:
    """
    FpML validation rule ird-8 - If the same party is specified as the payer and receiver, then different accounts must be specified.
    
    Parameters 
    ----------
    trade : Trade
    
    accounts : Account
    
    Returns
    -------
    success : boolean
    
    """
    self = inspect.currentframe()
    
    
    success = rosetta_resolve_attr(self, "success")
    
    
    return success

sys.modules[__name__].__class__ = create_module_attr_guardian(sys.modules[__name__].__class__)
